vunit/ui.py

```diff
@@ -393,6 +394,46 @@ def add_external_library(self, library_name, path, vhdl_standard=None):
         self._project.add_library(library_name, abspath(path), vhdl_standard, is_external=True)
         return self.library(library_name)
 
+    def add_source_files_from_csv(self, project_csv_path, vhdl_standard=None, is_in_order=True):
+        """
+        Add a project configuration, mapping all the libraries and files
+
+        :param project_csv_path: path to csv project specification, each line contains the name
+                                 of the library and the path to one file 'lib_name,filename'
+                                 note that all filenames are relative to the parent folder of the
+                                 csv file
+        :param vhdl_standard: The VHDL standard used to compile file into this library,
+                              if None, the VUNIT_VHDL_STANDARD environment variable is used
+        :param is_in_order: The csv file contains the list of libraries and files in the 
+                            correct compilation order, this overrides the automatic Vunit
+                            compilation dependencies
+        :returns: A list of files (:class `.SourceFileList`) that were added
+
+        """
+        if vhdl_standard is None:
+            vhdl_standard = self._vhdl_standard
+
+        libs = set()
+        files = SourceFileList(list())
+
+        with open(project_csv_path) as csv_path_file:
+            file_before = None
+            for row in csv.reader(csv_path_file):
+                if len(row) == 2:
+                    lib_name = row[0].strip()
+                    no_normalized_file = row[1].strip()
+                    file_name_ = normpath(join(dirname(project_csv_path), no_normalized_file))
+                    lib = self.library(lib_name) if lib_name in libs else self.add_library(lib_name)
+                    libs.add(lib_name)
+                    file_ = lib.add_source_file(file_name_, no_parse=is_in_order)
```

Comment:
No parse will also inhibit any test_benches/tests from being detected from the source file.

Reply:
When sigasi generates the csv file, we resolve all dependencies in a way that the first file is a dependency for the next one, for example
lib, example.vhd
lib2, example2.vhd  // This one depends on lib, example.vhd
lib, example3.vhd    // This one depends on lib2, example2.vhd
lib3, example4.vhd  // This one depends on lib, example3.vhd

Reply:
Dependencies are never linear they are a tree. To force them to be linear creates a lot of unnecessary re-compilation.

Reply:
Consider the case with 200 files with 20 groups of 10 files. Each group has a linear dependency but the groups are independent. If you can only express a linear order some group will be first and some group will be last. If a file in the first group is modified it causes 200 files to be recompiled instead of only 10.

Reply:
Sigasi for sure has a dependency tree internally. It is only when you need to generate a sequence of compile commands for the initial compile that it is useful to generate an ordered list of files. For subsequent *incremental* compiles which is a big feature of VUnit a new ordered list of compile commands has to generated based on the files changes and the dependency tree.